controllers/machine_controller_phases.go

```diff
@@ -239,6 +239,15 @@ func (r *MachineReconciler) reconcileInfrastructure(ctx context.Context, m *clus
 		}
 	}
 
+	// Get and set Status.InfrastructureCapabilities from the infrastructure provider.
+	err = util.UnstructuredUnmarshalField(infraConfig, &m.Status.InfrastructureCapabilities, "status", "capabilities")
```

Comment:
I'm calling it just capabilities on the infra record, as infrastructureCapabilities seems redundant on that object.